### PR TITLE
sum of divisors added

### PR DESCRIPTION
**Pull request by tk315**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

article on sum of divisors added

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#84

### CODE REVIEW

File: docs/mathematics/sum-of-divisors.mdx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+}
+```
+</TabItem>
+
+<TabItem value="py">
+
+```python   
+import math
+
+def sum_of_divisors(n):
```
Comment (atarax665) [inline docs/mathematics/sum-of-divisors.mdx:114]: Add proper comments where needed.

File: docs/mathematics/sum-of-divisors.mdx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+The sum of the divisors is: 91
+```
+
+## Complexity Analysis
+
+`n` is the given number.
+
+- Time Complexity:
+
+  - **Best Case**: O (sqrt(n))
```
Comment (atarax665) [inline docs/mathematics/sum-of-divisors.mdx:154]:
```suggestion
  - **Best Case**: `O (sqrt(n))`
```

File: docs/mathematics/sum-of-divisors.mdx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  - **Average Case**: O (sqrt(n))
```
Comment (atarax665) [inline docs/mathematics/sum-of-divisors.mdx:155]:
```suggestion
  - **Average Case**: `O(sqrt(n))`
```

File: docs/mathematics/sum-of-divisors.mdx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  - **Worst Case**: O (sqrt(n))
```
Comment (atarax665) [inline docs/mathematics/sum-of-divisors.mdx:156]:
```suggestion
  - **Worst Case**: `O(sqrt(n))`
```

File: docs/mathematics/sum-of-divisors.mdx [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+- Space Complexity: O (1)
```
Comment (atarax665) [inline docs/mathematics/sum-of-divisors.mdx:158]:
```suggestion
- Space Complexity: `O(1)`
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: docs/mathematics/sum-of-divisors.mdx (hunk 1 above)
Comment (tk315) [reply docs/mathematics/sum-of-divisors.mdx:114]: done

File: docs/mathematics/sum-of-divisors.mdx (hunk 2 above)
Comment (tk315) [reply docs/mathematics/sum-of-divisors.mdx]: done

File: docs/mathematics/sum-of-divisors.mdx (hunk 3 above)
Comment (tk315) [reply docs/mathematics/sum-of-divisors.mdx]: done

File: docs/mathematics/sum-of-divisors.mdx (hunk 4 above)
Comment (tk315) [reply docs/mathematics/sum-of-divisors.mdx]: done

File: docs/mathematics/sum-of-divisors.mdx (hunk 5 above)
Comment (tk315) [reply docs/mathematics/sum-of-divisors.mdx]: done

**Comment by tk315** [comment]
Please review @ankitaggarwal23 @HarshCasper @iamrajiv @atarax665

File: docs/mathematics/sum-of-divisors.mdx [hunk 6]
```diff
@@ -0,0 +1,166 @@
+---
+title: Sum of Divisors
+slug: /mathematics/sum-of-divisors
+---
+
+## Introduction
+
+Sum of divisors refers to the sum of numbers in the range `1 to n` which perfectly divides `n`.
```
Comment (HarshCasper) [inline docs/mathematics/sum-of-divisors.mdx:8]: Have a more detailed introduction. Usually follow this pattern while introducing what you are writing about: 
- Introduce the topic
- Why its relevant
- Prompt the reader to go next

File: docs/mathematics/sum-of-divisors.mdx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+## Explanation 
+
+We will make a sum variable and initialize it `0`. Then we will make a variable `i` and iterate it on the range `[1, sqrt(n)]` and will check whether `n` is divisible by `i` or not. If `n` is divisible by `i` and not equal to `sqrt(n)` we will add `i` and `n/i` to sum as both `i` and `n/i` will be the divisor of `n` because `i*(n/i) = n` and `i` lies on the left side of `sqrt(n)` and `n/i` on the right. If `n` is divisible by `i` and is equal to `sqrt(n)` we will only add `i` to sum as both `i` and `n/i` will be the same. 
+
```
Comment (HarshCasper) [inline docs/mathematics/sum-of-divisors.mdx:13]: Make it more intuitive and readable with normal English text and not jargons. Use the `Algorithm` section to detail about the formula and steps

File: docs/mathematics/sum-of-divisors.mdx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+num = 9
+i=2
+9 is not divisible by i
+
+num = 9
+i=3
+divisors = 3 , 9/3=3
+sum = sum + 3 = 13
+Here we will increase the sum by i only as i is equal to sqrt(num) 
+```
```
Comment (HarshCasper) [inline docs/mathematics/sum-of-divisors.mdx:34]: Use only two example. Also order them in a list

File: docs/mathematics/sum-of-divisors.mdx [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+<TabItem value="cpp">
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+long long int sum_of_divisors(int Number)
+{
+    /*Sum of Divisors may not fit in int range
+    so better take long long int */
```
Comment (HarshCasper) [inline docs/mathematics/sum-of-divisors.mdx:67]: Give a line break after a comment

File: docs/mathematics/sum-of-divisors.mdx [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+</TabItem>
+
+<TabItem value="py">
+
+```python   
+import math
+
+def sum_of_divisors(n):
+    i = 1
+    #Initilizing sum as 0
```
Comment (HarshCasper) [inline docs/mathematics/sum-of-divisors.mdx:116]: Typo here

File: docs/mathematics/sum-of-divisors.mdx [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+#Printing Output
+print("The sum of the divisors is: {} ".format(sum_of_divisors(n)))
+```
+</TabItem>
+
+</Tabs>
+
+## Sample Input and Output
+
+### Input :
```
Comment (HarshCasper) [inline docs/mathematics/sum-of-divisors.mdx:143]: Use Heading-4

File: docs/mathematics/sum-of-divisors.mdx [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+
+```
+Enter the number: 36
+```
+### Output :
```
Comment (HarshCasper) [inline docs/mathematics/sum-of-divisors.mdx:148]: Use Heading-4

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]

**Review by iamrajiv — APPROVED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**